Keys_in_react lesson: Fix typos and grammar mistakes (#25826)
TheOdinProject/curriculum: react/getting_started_with_react/keys_in_react.md

### Introduction

In this lesson, we will cover keys in React. Keys are special props for our components and we'll learn why they are used.

> ### Lesson Overview

Reviewer: We use "Sentence case" as per the [Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#headings).

```suggestion
### Lesson overview
```

Author: Hi, I made sure to check previous lessons on this but I've found that all other lessons use `Lesson Overview`, even in JS courses. 
![image](https://github.com/TheOdinProject/curriculum/assets/85281165/cc66ad77-139d-4c24-bb55-e8d7f9dbf520)

Reviewer 2: you're absolutely right and the sentence case is a recent change, hence why we try to catch these on PR's. Would you be willing to change the headers in this lesson?

Author: I didn't know it was a recent change. Yes, I'll change all the headers in this lesson.

}
~~~

> <span id="index-as-key">Additionally, if the list is static and will remain unchanged throughout the application's life, we can use the array index as a key. Though you should be careful when working with `index` as a key.</span>

Reviewer: Should we expand on the why here?

Reviewer 2: I was just going through this lesson and agree it would be nice to add "why we should be careful when working with `index` as a key"

Reviewer 2: what do you reckon @xandora ?

Reviewer 2: @Eveieve did you see the above? 

Some notes: 
- the headers should all be fixed already with the NodeJS path being fully updated to sentence-case last week
- I think it's good to proceed with the above change; add a note that the `why` will be covered in the assignment section. 
- there's a merge conflict to look into before we can merge it :)

Author: @ManonLef Hi, I left a comment above but I think there was a problem with this 'pending' next to my name. I think that's what's causing the merge conflict. 

![image](https://github.com/TheOdinProject/curriculum/assets/85281165/0871975a-a5ef-4284-9749-f9582ccee972)

I will post my suggestion here again to answer your second question: 

```suggestion
<span id="index-as-key">Additionally, if you're sure the list will remain unchanged throughout the application's life,  you can use the array index as a key. But this is [not recommended](https://react.dev/learn/rendering-lists#why-does-react-need-keys) since it can lead to confusing bugs if the list *does* change with any items deleted, inserted or rearranged. 
```
To your first question: I didn't change any headers for this PR, did you mean to just let me know about it?

Reviewer 2: I think I mentioned it because I asked you to change them earlier somewhere in this PR. 
If it had `pending` that's probably a review you started but didn't submit :) 

When I referred to `the above` your `pending` wasn't there yet so I meant the discussion me and Xandora had about it:

![Screenshot 2023-07-31 at 20 42 24](https://github.com/TheOdinProject/curriculum/assets/81025586/ba8f272b-7154-44b8-b538-736022d327aa)

The resource is shared later in the lesson so linking to it here as well doesn't feel like the best thing to do.

Reviewer 2: I like it! I'm just not sure about the last sentence:

> More on why this is will be covered in the reading later.

How do you feel about:

> You will learn more about this in the assignment section's linked articles. 

?

Reviewer 2: You can commit your own actually. I will do it now and do a final suggestion. Sorry about that

> <span id="index-as-key">Additionally, if you're sure the list will remain unchanged throughout the application's life,  you can use the array index as a key. But this is not recommended, since it can lead to confusing bugs if the list *does* change with any items deleted, inserted or rearranged. You will learn more about this in the assignment section's linked articles.</span>

Reviewer 2: ```suggestion
<span id="index-as-key">Additionally, if you're sure the list will remain unchanged throughout the application's life,  you can use the array index as a key. However, this is not recommended since it can lead to confusing bugs if the list changes when items are deleted, inserted, or rearranged. You will learn more about this in the assignment section's linked article.</span>
```

Reviewer 2: I should have paid better attention when I gave the previous feedback. I'm sorry about coming with yet another suggestion. Thanks for your time!
